### PR TITLE
[GLUTEN-6437][BUILD] Fix vcpkg setup-build-dependens.sh for centos

### PR DESCRIPTION
**Pull request by wecharyu**
## What changes were proposed in this pull request?

Replace the mirrorlist repository of centos with vault.

(Fixes: \#6437)

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)
Test the vcpkd setup scripty on the centos7/8 container:
```bash
# run container
podman run --name wechar_gluten --cap-add=SYS_PTRACE --cap-add=AUDIT_WRITE --pids-limit 0 -it --rm  -w /root/workspace --network host --entrypoint bash centos:7

# download gluten source code and run setup script
gluten/dev/vcpkg/setup-build-depends.sh
```

### CODE REVIEW

File: dev/vcpkg/setup-build-depends.sh [hunk 1]
```diff
@@ -45,21 +45,54 @@ install_gcc9_from_source() {
...
+    make -j$(nproc)
+    make install
+    echo "export PATH=${binutils_install_dir}/bin:\$PATH" > /etc/profile.d/binutils.sh && source /etc/profile
+    rm -rf ${binutils_version}* && cd
+}
+
 install_centos_7() {
     export PATH=/usr/local/bin:$PATH
 
-    yum -y install epel-release centos-release-scl
```
Comment (zhouyuan) [inline dev/vcpkg/setup-build-depends.sh:55]: I think we should keep it, and also update the mirror for epel repo
```
    rm /etc/yum.repos.d/CentOS-SCLo-scl.repo -f
    sed -i \
    -e 's/^mirrorlist/#mirrorlist/' \
    -e 's/^#baseurl/baseurl/' \
    -e 's/mirror\.centos\.org/vault.centos.org/' \
    /etc/yum.repos.d/CentOS-SCLo-scl-rh.repo

```

Comment (wecharyu) [reply dev/vcpkg/setup-build-depends.sh:55]: There seems no CentOS-SCLo-scl.repo and CentOS-SCLo-scl-rh.repo files in the default centos image.
And `vault` repository does not have `epel-release` `centos-release-scl` packages.

Comment (zhouyuan) [reply dev/vcpkg/setup-build-depends.sh:55]: I mean if we keep this line to install gcc from SCL repo
`yum -y install epel-release centos-release-scl`

Comment (zhouyuan) [reply dev/vcpkg/setup-build-depends.sh:55]: It does support SCL
https://github.com/apache/arrow/pull/43127

**Comment by wecharyu** [comment]
@zhouyuan pls take a look again, I have test on centos 7/8, it works fine now.

**Review by zhouyuan — APPROVED** [review]
👍

**Pull request merged**